recovery.go

```diff
@@ -33,6 +33,18 @@ func Recovery() HandlerFunc {
 
 // RecoveryWithWriter returns a middleware for a given writer that recovers from any panics and writes a 500 if there was one.
 func RecoveryWithWriter(out io.Writer) HandlerFunc {
+	return RecoveryWithWriterAndCallback(out, func(c *Context, err interface{}, brokenPipe bool) {
+		// If the connection is dead, we can't write a status to it.
+		if brokenPipe {
+			c.Error(err.(error)) // nolint: errcheck
+			c.Abort()
+		} else {
+			c.AbortWithStatus(http.StatusInternalServerError)
+		}
+	})
+}
+
+func RecoveryWithWriterAndCallback(out io.Writer, fn func(c *Context, err interface{}, brokenPipe bool)) HandlerFunc {
```

Comment:
Please add a comment block for this function.

Reply:
I did it